File: test/com/peeekay/aoc2023/java/Day04Test.java
package com.peeekay.aoc2023.java;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;

import static org.junit.jupiter.api.Assertions.assertEquals;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class Day04Test {
    final AOCPuzzle day = new Day04(true);

    @Test
    void part1() {
        day.solve();
        assertEquals(13L, day.part1());
    }

    @Test
    void part2() {
        day.solve();
        assertEquals(30L, day.part2());
    }

}

File: aoc2022/kotlin/src/test/kotlin/Day04Test.kt
package com.peeekay.aoc2022.kotlin

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("Day 04")
internal class Day04Test {

    @Test
    fun `Part1 with test data`() {
        val day = Day04(isTest = true)
        val answer = day.partOne()

        assertEquals(2, answer)
    }

    @Test
    fun `Part2 with test data`() {
        val day = Day04(isTest = true)
        val answer = day.partTwo()

        assertEquals(4, answer)
    }
}